chainercv/links/model/deeplab/xception.py

```diff
@@ -0,0 +1,155 @@
+            The default value is :obj:`conv`.
+        activ_first (boolean): If :obj:`True`, activation function is
+            applied first in this block.
+            The default value is :obj:`True`
+        bn_kwargs (dict): Keyword arguments passed to initialize
+            :class:`chainer.links.BatchNormalization`.
+
+    """
+
+    def __init__(self, in_channels, depthlist, stride=1, dilate=1,
+                 skip_type='conv', activ_first=True, bn_kwargs={},
+                 dw_activ_list=[None, None, None],
+                 pw_activ_list=[F.relu, F.relu, None]):
+        super(XceptionBlock, self).__init__()
+        self.skip_type = skip_type
+        self.activ_first = activ_first
+        self.separable2_activ = pw_activ_list[1]
+
+        with self.init_scope():
+            self.separable1 = SeparableConv2DBNActiv(
+                in_channels, depthlist[0], 3, 1,
+                dilate, dilate, nobias=True, bn_kwargs=bn_kwargs,
+                dw_activ=dw_activ_list[0], pw_activ=pw_activ_list[0])
+            self.separable2 = SeparableConv2DBNActiv(
+                depthlist[0], depthlist[1], 3, 1,
+                dilate, dilate, nobias=True, bn_kwargs=bn_kwargs,
+                dw_activ=dw_activ_list[1], pw_activ=F.identity)
+            self.separable3 = SeparableConv2DBNActiv(
+                depthlist[1], depthlist[2], 3, stride,
+                dilate, dilate, nobias=True, bn_kwargs=bn_kwargs,
+                dw_activ=dw_activ_list[2], pw_activ=pw_activ_list[2])
+            if skip_type == 'conv':
+                self.conv = Conv2DBNActiv(
+                    in_channels, depthlist[2], 1, activ=F.identity,
+                    nobias=True, stride=stride, bn_kwargs=bn_kwargs)
+
+    def __call__(self, x):
+        if self.activ_first:
+            h = F.relu(x)
+        else:
+            h = x
+
+        h = self.separable1(h)
+        h = self.separable2(h)
+        separable2 = h
+        h = self.separable2_activ(h)
+        h = self.separable3(h)
+
+        if self.skip_type == 'conv':
+            skip = self.conv(x)
+            h = h + skip
+        elif self.skip_type == 'sum':
+            h = h + x
+        elif self.skip_type == 'none':
+            pass
+
+        if not self.activ_first:
+            h = F.relu(h)
+
+        return h, separable2
+
+
+class Xception65(chainer.Chain):
+
+    """Xception65 for backbone network of DeepLab v3+.
+
+    Unlike original Xception65, this follows implementation in deeplab v3
+    (https://github.com/tensorflow/models/tree/master/research/deeplab).
+    this returns lowlevel feature (an output of second convolution in second
+    block in entryflow) and highlevel feature (an output before final average
+    pooling in original).
+
+    Args:
+        bn_kwargs (dict): Keyword arguments passed to initialize
+            :class:`chainer.links.BatchNormalization`.
+
+    """
+
+    mean_pixel = [127.5, 127.5, 127.5]
```

Comment:
Can you change the variable name to `mean` so that the interface is consistent with `ResNet`?
This consistency would make it easier to swap different feature extractors.